Description: Upstream changes introduced in version 1:1.2.3.4.dfsg-3ubuntu1
 This patch has been created by dpkg-source during the package build.
 Here's the last changelog entry, hopefully it gives details on why
 those changes were made:
 .
 zlib (1:1.2.3.4.dfsg-3ubuntu1) maverick; urgency=low
 .
   * Merge from debian testing.  Remaining changes:
     - zlib.h: add z_off64_t and ensure that its mapped to the correct
       off_t or off64_t (fixes FTBFS in vips, gtk)
     - drop Conflicts on libc6-i386 (<= 2.9-18) because we do not have
       to do a /emul/ia32-linux -> /usr/lib32 transition
 .
 The person named in the Author field signed this changelog entry.
Author: Didier Roche <didrocks@ubuntu.com>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- zlib-1.2.3.4.dfsg.orig/zlib.h
+++ zlib-1.2.3.4.dfsg/zlib.h
@@ -1435,12 +1435,20 @@ ZEXTERN int ZEXPORT inflateBackInit_ OF(
         inflateBackInit_((strm), (windowBits), (window), \
                                             ZLIB_VERSION, sizeof(z_stream))
 
-#ifdef _LARGEFILE64_SOURCE
+#if defined(_LARGEFILE64_SOURCE)
+    #define z_off64_t off64_t
+#else 
+    #if _FILE_OFFSET_BITS == 64
+        #define z_off64_t off_t
+    #endif
+#endif
+
+#if defined(_LARGEFILE64_SOURCE) || _FILE_OFFSET_BITS == 64
    ZEXTERN gzFile ZEXPORT gzopen64 OF((const char *, const char *));
-   ZEXTERN off64_t ZEXPORT gzseek64 OF((gzFile, off64_t, int));
-   ZEXTERN off64_t ZEXPORT gztell64 OF((gzFile));
-   ZEXTERN uLong ZEXPORT adler32_combine64 OF((uLong, uLong, off64_t));
-   ZEXTERN uLong ZEXPORT crc32_combine64 OF((uLong, uLong, off64_t));
+   ZEXTERN z_off64_t ZEXPORT gzseek64 OF((gzFile, z_off64_t, int));
+   ZEXTERN z_off64_t ZEXPORT gztell64 OF((gzFile));
+   ZEXTERN uLong ZEXPORT adler32_combine64 OF((uLong, uLong, z_off64_t));
+   ZEXTERN uLong ZEXPORT crc32_combine64 OF((uLong, uLong, z_off64_t));
 #endif
 
 #if !defined(ZLIB_INTERNAL) && _FILE_OFFSET_BITS == 64
